### PR TITLE
[WFLY-19567]: Removing ConcurrentHashMap usage

### PR DESCRIPTION
**Pull request by ehsavoie**
Thanks for submitting your Pull Request!

Please delete this text, and add a link to the Jira issue solved by this PR.

If this PR is not for the 'main' branch you must add a link to the equivalent change in 'main'.

Remember to use the Jira issue ID in the PR title and any commits.

____

<--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->

> Wildfly issue links:
> * [WFLY-19567](https://issues.redhat.com/browse/WFLY-19567)

<--- END OF WILDFLY GITHUB BOT REPORT --->

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

### CODE REVIEW

File: messaging-activemq/injection/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java [hunk 1]
```diff
@@ -23,12 +23,21 @@
...
 
-    private final Map<String, JMSContext> contexts = new ConcurrentHashMap<>();
+    private final transient ReentrantLock lock = new ReentrantLock();
+    private final transient Map<String, JMSContext> contexts = new HashMap<>();
 
-    JMSContext getContext(String injectionPointId, JMSInfo info, ConnectionFactory connectionFactory) {
-        return contexts.computeIfAbsent(injectionPointId, key -> {
-            return createContext(info, connectionFactory);
-        });
+    synchronized JMSContext getContext(String injectionPointId, JMSInfo info, ConnectionFactory connectionFactory) {
```
Comment (bstansberry) [inline messaging-activemq/injection/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java:29]: Not synchronized. :)

File: messaging-activemq/injection/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java [hunk 2]
```diff
@@ -23,12 +23,21 @@
...
+        try {
+            JMSContext context = contexts.get(injectionPointId);
+            if (context == null) {
+                context = createContext(info, connectionFactory);
+                contexts.put(injectionPointId, context);
+            }
+            return context;
+        } finally {
+            lock.unlock();
+        }
```
Comment (bstansberry) [inline messaging-activemq/injection/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java:40]: If this method is called a lot you can stick with ConcurrentHashMap, do a get call, and fall back to this lock stuff if it returns null.

File: messaging-activemq/injection/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java (hunk 1 above)
Comment (ehsavoie) [reply messaging-activemq/injection/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java:29]: yes sorry, leftover

File: messaging-activemq/injection/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java (hunk 2 above)
Comment (bstansberry) [reply messaging-activemq/injection/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java:40]: @ehsavoie You switched back to ConcurrentHashMap but you're still doing the initial read with the lock held.

Basically I'm suggesting double checked locking.

**Review by bstansberry — APPROVED** [review]
@ehsavoie This LGTM. Can you move it out of Draft state?

**Comment by bstansberry** [comment]
Thanks @ehsavoie

**Pull request merged**